tests/test_app/TestApp/ServiceProvider/PersonServiceProvider.php

```diff
@@ -0,0 +1,22 @@
+<?php
+declare(strict_types=1);
+
+namespace TestApp\ServiceProvider;
+
+use Cake\Core\ContainerInterface;
+use Cake\Core\ServiceProvider;
+
+class PersonServiceProvider extends ServiceProvider
+{
+    protected $provides = ['boot', 'sally'];
```

Comment:
I don't like this being a property in League DIC, not very intuitive. Can we have a method for this?

Reply:
Based on the docs https://container.thephpleague.com/3.x/service-providers/  it doesn't seem there's any alternative to populating this array. So we would have to add a method ourselves if needed.

Reply:
That's what I had in mind.

Reply:
Would the method return the list of services provides? That will end up running a bit slower than a property would as the array would be re-allocated many times during a request.